packages/daemon/src/pet-store.js

```diff
@@ -132,6 +107,35 @@ export const makePetStoreMaker = (filePowers, locator) => {
       // TODO consider tracking historical pet names for formulas
     };
 
+    /**
+     * @param {string} petName
+     * @param {string} formulaIdentifier
+     */
+    const write = async (petName, formulaIdentifier) => {
+      assertValidName(petName);
+      if (!validFormulaPattern.test(formulaIdentifier)) {
+        throw new Error(`Invalid formula identifier ${q(formulaIdentifier)}`);
+      }
+
+      if (petNames.has(petName)) {
+        await remove(petName);
```

Comment:
This `await` probably introduces a race condition. Can we avoid that by inlining the synchronous behavior of `remove`?

Reply:
Or rather, an interleaving hazard for concurrent writes.